lib/rpcserver/tarpc_imp.c

```diff
@@ -1881,6 +1881,27 @@ TARPC_FUNC(statvfs, {},
 )
 #endif /* HAVE_SYS_STATVFS_H */
 
+/*-------------- readlink() --------------------------------*/
+TARPC_FUNC(readlink, {},
+{
+    char *buf;
+
+    TARPC_ENSURE_NOT_NULL(path);
+
+    buf = (char *)calloc(sizeof(in->bufsize), 1);
```

Comment:
`TE_ALLOC`, please. And the result of memory allocation shall never be explicitly cast to a specific type.